### PR TITLE
Add Weaviate integration

### PR DESCRIPTION
**Pull request by hunteritself**
Add Weaviate integration, features include:
1. Initiate the required environment and connect to Weaviate vector database;
2. Create a class;
3. Delete a class;
4. Add data;
5. Make similarity-based queries.

### CODE REVIEW

**Comment by jarulraj** [comment]
Nice PR! Can we also add support for label-based filtering? -- https://weaviate.io/developers/weaviate/search/filters. Like searching only among a subset of documents that satisfy some label value (e.g., label = 'cs450'). We can discuss more via Slack.

**Comment by hunteritself** [comment]
Sure Professor. I'll add this "label-based filtering" feature soon.

File: evadb/third_party/vector_stores/weaviate.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+        # Get the API key.
+        self._api_key = kwargs.get("WEAVIATE_API_KEY")
+
+        if not self._api_key:
+            self._api_key = os.environ.get("WEAVIATE_API_KEY")
+
+        assert (
+            self._api_key
+        ), "Please set your Weaviate API key in evadb.yml file (third_party, weaviate_api_key) or " \
```
Comment (xzdandy) [inline evadb/third_party/vector_stores/weaviate.py:41]: There is no evadb.yml anymore. Could you update the message? Thanks.

**Comment by xzdandy** [comment]
Hi @hunteritself, could you add a test case and documentation please? 
- For documentation, you can use https://evadb.readthedocs.io/en/stable/source/reference/vector_databases/index.html as a reference.
- For test case, you can use https://github.com/georgia-tech-db/evadb/blob/staging/test/integration_tests/long/test_similarity.py this as a reference.

File: evadb/third_party/vector_stores/weaviate.py (hunk 1 above)
Comment (aaronyang-ai) [reply evadb/third_party/vector_stores/weaviate.py]: Thanks Andy! I've reorganized my code structure for the weaviate integration and added the relevant documentation and test.

**Comment by xzdandy** [comment]
You can fix the linter by running `bash script/test/test.sh -m LINTER`

**Comment by aaronyang-ai** [comment]
Hi Andy,

I have fixed all the code formatting errors suggested. The local test `bash script/test/test.sh -m LINTER` has passed. But the CircleCI test on GitHub still failed, showing test/integration_tests/long/test_similarity.py has incorrect formatting. Could you please help take a look? Thank you.

**Review by xzdandy — APPROVED** [review]

**Pull request merged**